src/Magnum/Platform/WindowlessEglApplication.h

```diff
@@ -276,16 +276,21 @@ class WindowlessEglContext::Configuration {
          * @m_since_latest
          *
          * When set, the created context will share a subset of OpenGL objects
-         * with @p context, instead of being independent. Many caveats and
+         * with @p context whose associated display is @p display, instead of being independent. 
+         * Many caveats and
          * limitations apply to shared OpenGL contexts, please consult the
          * OpenGL specification for details. Default is `EGL_NO_CONTEXT`, i.e.
          * no sharing.
+         *
+         * @note @p display must be provided to avoid errors on destruction, since 
+         * the display is shared among shared-context on EGL.
          * @see @ref WindowlessEglContext::glContext(),
          *      @ref WindowlessEglApplication::glContext()
          * @requires_gles Context sharing is not available in WebGL.
          */
-        Configuration& setSharedContext(EGLContext context) {
+        Configuration& setSharedContext(EGLContext context, EGLDisplay display) {
             _sharedContext = context;
```

Comment:
I'd add a 

```cpp
CORRADE_ASSERT((context == EGL_NO_CONTEXT) == (display == EGL_NO_DISPLAY),
    "Platform::WindowlessEglContext::Configuration::setSharedContext(): either both context has to be valid or both invalid.", *this);
```

here, moving the implementation to the cpp, and drop the same check inside the `WindowlessEglContext` constructor -- better check this as early as possible, and since it's a programmer error, it should be an assert and not an Error.

Reply:
Corrected !